src/include/gurt/telemetry_common.h

```diff
@@ -18,6 +18,52 @@
 #define D_TM_SHARED_MEMORY_KEY		0x10242048
 #define D_TM_SHARED_MEMORY_SIZE		(1024 * 1024)
 
+#define D_TM_MIN(stats, name) \
+	stats->dtm_min.min_##name
+
+#define D_TM_MAX(stats, name) \
+	stats->dtm_max.max_##name
+
+#define D_TM_PRINT_STATS(s, stats, name, fmt) \
+	do {								       \
+		fprintf(s, " min: %"#fmt " max: %"#fmt " mean: %lf size: %"    \
+			PRIu64, D_TM_MIN(stats, name), D_TM_MAX(stats, name),  \
+			stats->mean, stats->sample_size);		       \
+		if (stats->sample_size > 2)				       \
+			fprintf(s, " std dev: %lf", stats->std_dev);	       \
+	} while (0)
+
+#define D_TM_COMPUTE_MIN(value, node, name)				       \
+	do {								       \
+		if ((value) < node->dtn_metric->dtm_stats->dtm_min.min_##name) \
+			node->dtn_metric->dtm_stats->dtm_min.min_##name =      \
+			(value);					       \
+	} while (0)
+
+#define D_TM_COMPUTE_MAX(value, node, name)				       \
+	do {								       \
+		if ((value) > node->dtn_metric->dtm_stats->dtm_max.max_##name) \
+			node->dtn_metric->dtm_stats->dtm_max.max_##name =      \
+			(value);					       \
+	} while (0)
+
+#define D_TM_COMPUTE_MEAN(var, node, value)				       \
+	(var) = node->dtn_metric->dtm_stats->mean +			       \
+		(((value) - node->dtn_metric->dtm_stats->mean) /	       \
+		node->dtn_metric->dtm_stats->sample_size);		       \
+
+#define D_TM_VALUE(node)						       \
+	((node->dtn_type & D_TM_DURATION) ?				       \
+		(node->dtn_metric->dtm_data.tms[0].tv_sec +		       \
+		(node->dtn_metric->dtm_data.tms[0].tv_nsec / 1E9)) :	       \
+			(node->dtn_type == D_TM_GAUGE) ?		       \
+				node->dtn_metric->dtm_data.value : 0)
+
+#define D_TM_COMPUTE_SUM_OF_SQUARES(var, node, value)			       \
```

Comment:
(style)  Macros with multiple statements should be enclosed in a do - while loop
(style)  macros should not use a trailing semicolon